src/main/java/mat/model/clause/CQLLibrary.java

```diff
@@ -346,4 +351,22 @@ public void setFhirVersion(String fhirVersion) {
         this.fhirVersion = fhirVersion;
     }
 
+    @Transient
+    public String getLibraryXMLAsString() {
+        String xml = "";
+        if (cqlXML != null) {
+            xml = new String(toByteArray(cqlXML));
+        }
+        return xml;
+    }
+
+    public void setLibraryXMLAsByteArray(String xml) {
+        if (null != xml) {
+            byte[] xmlByteArr = xml.getBytes();
+            this.cqlXML = Hibernate.getLobCreator(HibernateConf.getHibernateSession()).createBlob(xmlByteArr);
+        } else {
+            this.cqlXML = null;
```

Comment:
Did you try to get rid of an explicit LOB object? I believe that JPA should be able to handle byte array AS IS and a LOB instance will be created implicitly behind the scenes. That would reduce boiler plate code.